libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -1167,6 +1167,30 @@ bool AP_Arming::pre_arm_checks(bool report)
 
 bool AP_Arming::arm_checks(AP_Arming::Method method)
 {
+    if (method == AP_Arming::Method::RUDDER) {
+        if (get_rudder_arming_type() == AP_Arming::RudderArming::IS_DISABLED) {
+            //parameter disallows rudder arming/disabling
+            return false;
+        }
+
+        // only permit arming if the vehicle isn't being commanded to
+        // move via RC input
+        const RCMapper * rcmap = AP::rcmap();
+        if (rcmap == nullptr) {
+            check_failed(true, "No rcmap");
+            return false;
+        }
+        RC_Channel *c = rc().channel(rcmap->throttle() - 1);
+        if (c == nullptr) {
```

Comment:
Can we collapse `No rcmap` to be the same check failed as `No throttle channel`? It's still very accurate, it's a firmware specific thing, and it saves us the error string in all cases....